controllers/operatorconfigmap_controller.go

```diff
@@ -591,3 +598,62 @@ func labelClientOperatorSubscription(c *OperatorConfigMapReconciler) error {
 	c.log.Info("successfully labelled ocs-client-operator subscription")
 	return nil
 }
+
+func (c *OperatorConfigMapReconciler) reconcileSubscription() error {
+
+	storageClients := &v1alpha1.StorageClientList{}
+	if err := c.list(storageClients); err != nil {
+		return fmt.Errorf("failed to list storageclients: %v", err)
+	}
+
+	var desiredChannel string
+	for idx := range storageClients.Items {
+		// empty if annotation doesn't exist or else gets desired channel
+		channel := storageClients.Items[idx].GetAnnotations()[utils.DesiredSubscriptionChannelAnnotationKey]
+		// skip clients with no/empty desired channel annotation
+		if channel != "" {
+			// check if we already established a desired channel
+			if desiredChannel == "" {
+				desiredChannel = channel
+			}
+			// check for agreement between clients
+			if channel != desiredChannel {
+				desiredChannel = ""
+				// two clients didn't agree for a same channel and no need to continue further
+				break
+			}
+		}
+	}
+	if desiredChannel == "" {
+		// no desired channel specified or no agreement between clients
+		return nil
+	}
+
+	subscriptions := &opv1a1.SubscriptionList{}
+	err := c.list(subscriptions, client.InNamespace(c.OperatorNamespace), client.MatchingLabels{subscriptionLabelKey: subscriptionLabelValue}, client.Limit(1))
+	if err != nil {
+		return fmt.Errorf("failed to list subscription for ocs-client-operator using labels: %v", err)
+	}
+	if len(subscriptions.Items) != 1 {
+		return fmt.Errorf("failed to load subscription for ocs-client-operator")
+	}
```

Comment:
Should we consider it an error and halt all other reconciliation functions because we didn't find a subscription? In my eyes, it seems like an overkill. If there is no subscription we can assume the operator is not upgradable easily or at least not using OLM which means it will be really hard to get the version to ignore the restrictions.

Reply:
yes, makes sense, not having a subscription will not effect the running operator.

Reply:
done.